runtimes/rust/tests/migration.rs

```diff
@@ -437,3 +468,79 @@ fn test_resume_incorrect_artifact_name() {
         ErrorMatch::from_fail(&CoreError::CannotResumeService)
     );
 }
+
+#[test]
+fn test_resume_service_error() {
+    let (mut blockchain, _) = create_runtime();
+    let keypair = blockchain.as_ref().service_keypair().clone();
+
+    // Make withdrawal.
+    let amount = 10_000;
+    execute_transaction(
+        &mut blockchain,
+        keypair.withdraw(WithdrawalServiceV1::INSTANCE_ID, amount),
+    )
+    .unwrap();
+
+    // Stop running service instance.
+    execute_transaction(
+        &mut blockchain,
+        keypair.stop_service(
+            ToySupervisorService::INSTANCE_ID,
+            StopService {
+                instance_id: WithdrawalServiceV1::INSTANCE_ID,
+            },
+        ),
+    )
+    .unwrap();
+
+    // Make fast-forward migration to the WithdrawalServiceV2.
+    execute_transaction(
+        &mut blockchain,
+        keypair.migrate_service(
+            ToySupervisorService::INSTANCE_ID,
+            MigrateService {
+                instance_name: WithdrawalServiceV1::INSTANCE_NAME.to_owned(),
+                artifact: WithdrawalServiceV2.artifact_id(),
+            },
+        ),
+    )
+    .unwrap();
+
+    // Resume stopped service instance.
+    let actual_err = execute_transaction(
+        &mut blockchain,
+        keypair.resume_service(
+            ToySupervisorService::INSTANCE_ID,
+            ResumeService {
+                instance_id: WithdrawalServiceV2::INSTANCE_ID,
+                artifact: WithdrawalServiceV2.artifact_id(),
+                params: vec![1, 2, 3, 4],
+            },
+        ),
+    )
+    .unwrap_err();
+
+    // Verify the service instance after migration and unsuccessful resume.
+    let instance_state = blockchain
+        .snapshot()
+        .for_dispatcher()
+        .get_instance(WithdrawalServiceV1::INSTANCE_ID)
+        .unwrap();
+
+    assert_eq!(
+        instance_state.spec,
+        WithdrawalServiceV1.default_instance().instance_spec
+    );
+    assert_eq!(instance_state.status, Some(InstanceStatus::Stopped));
+    assert_eq!(
+        instance_state.data_version(),
+        &WithdrawalServiceV2.artifact_id().version
+    );
+
+    assert_eq!(
```

Comment:
Nit: Why is this check so far from `actual_err` instantiation? I'd suggest to place it right after instantiation.